### ISSUE TITLE
[BUG] launching a registered workflow with a `Union` type breaks the WebUI

### ISSUE THREAD

**Issue by alekpikl** [flyteorg/flyte#3839]
### Describe the bug

Minimal reproducible example:
```python
from typing import Union

from flytekit import task, workflow

@task()
def my_task(number: Union[int, str]):
    print(number)


@workflow
def my_work(number: Union[int, str] = 1):
    my_task(number=number)
```

Registering the workflow works: `pylyte register workflow.py` ✅.
Running the workflow directly works: `pylyte run --remote run workflow.py my_workflow` ✅.
Trying to run the registered workflow from the WebUI fails ❌ (see screenshot).

### Expected behavior

I expect an error at registration or in the WebUI if the workflow is broken. Currently, the WebUI errors out without a hint.

### Screenshots

The error in the WebUI:
<img width="479" alt="Bildschirmfoto 2023-07-06 um 09 56 33" src="https://github.com/flyteorg/flyte/assets/19668962/a60c0fc5-7510-4f56-9a19-0ccd1a97ecf2">

More information by pressing Fn12:
<img width="556" alt="Bildschirmfoto 2023-07-06 um 09 59 07" src="https://github.com/flyteorg/flyte/assets/19668962/42ed3457-e64e-42fc-bb0d-46e4a3261bf4">

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by eapolinario** [comment]
@alekpikl , can you confirm which version of flyteconsole you're running?

**Comment by eapolinario** [comment]
The functionality provided by the `file` utility is present in a library called `libmagic` which has [python bindings](https://pypi.org/project/python-libmagic/). 

We could hook `python-libmagic` up with the aliases [exported](https://github.com/flyteorg/flytekit/blob/30060324122760a57345722cea8adcb154875f49/flytekit/types/file/__init__.py#L60-L116) in the FlyteFile type transformers, more specifically in the [`to_literal`](https://github.com/flyteorg/flytekit/blob/master/flytekit/types/file/file.py#L325-L331).

**Comment by alekpikl** [comment]
>  can you confirm which version of flyteconsole you're running?

v1.6.0

**Comment by eapolinario** [comment]
@alekpikl , this was fixed in flyteconsole 1.8.3.

If you're using the helm charts to install Flyte, this went out in the Flyte 1.7.0 release.
